src/QMCWaveFunctions/LCAO/SoaLocalizedBasisSet.cpp

```diff
@@ -21,6 +21,58 @@
 
 namespace qmcplusplus
 {
+
+template<class COT, typename ORBT>
+void SoaLocalizedBasisSet<COT, ORBT>::createResource(ResourceCollection& collection) const
+{
+  for (int i = 0; i < LOBasisSet.size(); i++)
+    LOBasisSet[i]->createResource(collection);
+}
+template<class COT, typename ORBT>
+void SoaLocalizedBasisSet<COT, ORBT>::acquireResource(
+    ResourceCollection& collection,
+    const RefVectorWithLeader<SoaBasisSetBase<ORBT>>& basis_list) const
+{
+  auto& loc_bs_leader  = basis_list.template getCastedLeader<SoaLocalizedBasisSet<COT, ORBT>>();
+  auto& atom_bs_leader = loc_bs_leader.LOBasisSet;
+  const int num_ctr    = loc_bs_leader.LOBasisSet.size();
+  for (int i = 0; i < num_ctr; i++)
+  {
+    const auto atom_basis_list(extractLOBasisRefList(basis_list, i));
+    atom_bs_leader[i]->acquireResource(collection, atom_basis_list);
+  }
+}
+template<class COT, typename ORBT>
+void SoaLocalizedBasisSet<COT, ORBT>::releaseResource(
+    ResourceCollection& collection,
+    const RefVectorWithLeader<SoaBasisSetBase<ORBT>>& basis_list) const
+{
+  auto& loc_bs_leader  = basis_list.template getCastedLeader<SoaLocalizedBasisSet<COT, ORBT>>();
+  auto& atom_bs_leader = loc_bs_leader.LOBasisSet;
+  const int num_ctr    = loc_bs_leader.LOBasisSet.size();
+  for (int i = 0; i < num_ctr; i++)
+  {
+    const auto atom_basis_list(extractLOBasisRefList(basis_list, i));
+    atom_bs_leader[i]->releaseResource(collection, atom_basis_list);
+  }
+}
+template<class COT, typename ORBT>
+RefVectorWithLeader<COT> SoaLocalizedBasisSet<COT, ORBT>::extractLOBasisRefList(
```

Comment:
Since you are extracting the basis associated with a species, `extractOneSpeciesBasisRefList` is better than `extractLOBasisRefList`

Reply:
fixed